addons/pkg/util/cluster_variable_util.go

```diff
@@ -11,33 +11,44 @@ import (
 	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
+func ParseClusterVariableBool(cluster *clusterapiv1beta1.Cluster, variableName string) (bool, error) {
+	_, result, err := parseClusterVariable(cluster, variableName)
+	return result, err
+}
+
 func ParseClusterVariableString(cluster *clusterapiv1beta1.Cluster, variableName string) (string, error) {
+	result, _, err := parseClusterVariable(cluster, variableName)
+	return result, err
+}
+
+func parseClusterVariable(cluster *clusterapiv1beta1.Cluster, variableName string) (string, bool, error) {
```

Comment:
Wondering if we could use interface{} type for return value, and use reflect to handle it?

Reply:
Having multiple return types looks a bit weird.

Reply:
we can use golang reflection to handle that

Reply:
I agree on this

Reply:
changed, added a unit test for boolean as well, could you plz check? @maralavi